src/System.Globalization/tests/CompareInfo.cs

```diff
@@ -0,0 +1,424 @@
+
+        Assert.Throws<ArgumentOutOfRangeException>(() => ci.Compare("good", 0, badString, badOffset, CompareOptions.None));
+        Assert.Throws<ArgumentOutOfRangeException>(() => ci.Compare(badString, badOffset, "good", 0, CompareOptions.None));
+    }
+
+    [Theory]
+    [InlineData(null, 0, -1)]
+    [InlineData(null, -1, 0)]
+    [InlineData(null, -1, -1)]
+    [InlineData("", 0, -1)]
+    [InlineData("", -1, 0)]
+    [InlineData("", -1, -1)]
+    [InlineData("abc", 0, 4)]
+    [InlineData("abc", 4, 0)]
+    [InlineData("abc", 0, -1)]
+    [InlineData("abc", -1, 3)]
+    [InlineData("abc", 2, 2)]
+    public static void CompareArgumentOutOfRangeIndexAndCount(string badString, int badOffset, int badCount)
+    {
+        CompareInfo ci = CultureInfo.InvariantCulture.CompareInfo;
+
+        Assert.Throws<ArgumentOutOfRangeException>(() => ci.Compare("good", 0, 4, badString, badOffset, badCount));
+        Assert.Throws<ArgumentOutOfRangeException>(() => ci.Compare(badString, badOffset, badCount, "good", 0, 4));
+
+        Assert.Throws<ArgumentOutOfRangeException>(() => ci.Compare("good", 0, 4, badString, badOffset, badCount, CompareOptions.Ordinal));
+        Assert.Throws<ArgumentOutOfRangeException>(() => ci.Compare(badString, badOffset, badCount, "good", 0, 4, CompareOptions.Ordinal));
+    }
+
+    [Fact]
+    public static void CompareBadCompareOptions()
+    {
+        CompareInfo ci = CultureInfo.InvariantCulture.CompareInfo;
+
+        Assert.Throws<ArgumentException>(() => ci.Compare("a", "b", CompareOptions.Ordinal | CompareOptions.IgnoreWidth));
+        Assert.Throws<ArgumentException>(() => ci.Compare("a", "b", CompareOptions.OrdinalIgnoreCase | CompareOptions.IgnoreWidth));
+        Assert.Throws<ArgumentException>(() => ci.Compare("a", "b", (CompareOptions)(-1)));
+    }
+
+    [Theory]
+    [MemberData("IndexOfData")]
+    [ActiveIssue(846, PlatformID.AnyUnix)]
+    public static void IndexOf(string localeName, string source, string value, int expectedResult, CompareOptions options)
+    {
+        CompareInfo ci = CompareInfo.GetCompareInfo(localeName);
+
+        Assert.Equal(expectedResult, ci.IndexOf(source, value, options));
+
+        if (value.Length == 1)
+        {
+            Assert.Equal(expectedResult, ci.IndexOf(source, value[0], options));
+        }
+
+        if (options == CompareOptions.None)
+        {
+            Assert.Equal(expectedResult, ci.IndexOf(source, value));
+        }
+
+        if (value.Length == 1 && options == CompareOptions.None)
+        {
+            Assert.Equal(expectedResult, ci.IndexOf(source, value[0]));
+        }
+    }
+
+    [Fact]
+    [ActiveIssue(846, PlatformID.AnyUnix)]
+    public static void IndexOfMinusOneEverettCompatability()
+    {
+        CompareInfo ci  = CultureInfo.InvariantCulture.CompareInfo;
+
+        // This behavior was for Everett compatability.  We early outed for empty source strings
```

Comment:
Could we change the comment to list a known version number rather than a codename?